### PR TITLE
Add Multi-Region KMS Keymaker Keys

### PR DESCRIPTION
**Pull request by ryan-brown-sb**
- adds new multi-region KeyMaker primary module and keys
 - adds new multi-region KeyMaker replica module and keys
 - fixes https://github.com/18F/identity-devops/issues/4550

### CODE REVIEW

**Comment by ryan-brown-sb** [comment]
<details><summary><tt>terraform app/ryandbrown plan</tt></summary>

```

Terraform will perform the following actions:

  # aws_s3_object.version_info will be updated in-place
  ~ resource "aws_s3_object" "version_info" {
      ~ content                = <<-EOT
          - main_commit=2215fb646652094bd33139d829a5fb22b8fc0723
          - main_branch=HEAD
          + main_commit=616ab5949d7c9710292b79fad3f9c79f2a0c92d5
          + main_branch=ryandbrown/multiregion_kms
            main_version=pre-254
            private_commit=1c5713e54ad33daaf740d4d33d892c6a9ddf50a8
            private_branch=main
        EOT
        id                     = "terraform-app/version_info/ryandbrown.txt"
        tags                   = {}
      ~ version_id             = "6._HTZ3tTzpjZnf4nStHx6Djww7spNc5" -> (known after apply)
        # (11 unchanged attributes hidden)
    }

  # module.athena_events_log_database.aws_lambda_function.log_processor_lambda[0] will be updated in-place
  ~ resource "aws_lambda_function" "log_processor_lambda" {
        id                             = "ryandbrown-events-log-processor"
      ~ last_modified                  = "2023-04-13T19:42:58.000+0000" -> (known after apply)
      ~ source_code_hash               = "ayg5fF83omtGb1z3ua/3oLD6mdho4p6PXTxG323lGXM=" -> "9HLTOP5xYdgqRQLJjCviFr6ZoaC+jZH6om8/E9jz2Mc="
        tags                           = {}
        # (21 unchanged attributes hidden)

        # (2 unchanged blocks hidden)
    }

  # module.kms_keymaker_replica_ue1.data.aws_iam_policy_document.kms_events_topic_policy will be read during apply
  # (config refers to values not yet known)
 <= data "aws_iam_policy_document" "kms_events_topic_policy" {
      + id        = (known after apply)
      + json      = (known after apply)
      + policy_id = "kms_ct_sqs"

      + statement {
          + actions   = [
              + "SNS:Publish",
            ]
          + effect    = "Allow"
          + resources = [
              + (known after apply),
            ]
          + sid       = "KMSCTSQS"

          + principals {
              + identifiers = [
                  + "events.amazonaws.com",
                ]
              + type        = "Service"
            }
        }
    }

  # module.kms_keymaker_replica_ue1.aws_cloudwatch_event_rule.decrypt will be created
  + resource "aws_cloudwatch_event_rule" "decrypt" {
      + arn            = (known after apply)
      + description    = "Capture decryption events"
      + event_bus_name = "default"
      + event_pattern  = (known after apply)
      + id             = (known after apply)
      + is_enabled     = true
      + name           = "ryandbrown-decryption-events"
      + name_prefix    = (known after apply)
      + tags_all       = (known after apply)
    }

  # module.kms_keymaker_replica_ue1.aws_cloudwatch_event_target.sns will be created
  + resource "aws_cloudwatch_event_target" "sns" {
      + arn            = (known after apply)
      + event_bus_name = "default"
      + id             = (known after apply)
      + rule           = "ryandbrown-decryption-events"
      + target_id      = "ryandbrown-sns"
    }

  # module.kms_keymaker_replica_ue1.aws_kms_alias.login-dot-gov-keymaker-alias will be created
  + resource "aws_kms_alias" "login-dot-gov-keymaker-alias" {
      + arn            = (known after apply)
      + id             = (known after apply)
      + name           = "alias/ryandbrown-login-dot-gov-keymaker"
      + name_prefix    = (known after apply)
      + target_key_arn = (known after apply)
      + target_key_id  = (known after apply)
    }

  # module.kms_keymaker_replica_ue1.aws_kms_replica_key.login-dot-gov-keymaker-multi-region-replica will be created
  + resource "aws_kms_replica_key" "login-dot-gov-keymaker-multi-region-replica" {
      + arn                                = (known after apply)
      + bypass_policy_lockout_safety_check = false
      + deletion_window_in_days            = 30
      + description                        = "ryandbrown-login-dot-gov-keymaker-multi-region"
      + enabled                            = true
      + id                                 = (known after apply)
      + key_id                             = (known after apply)
      + key_rotation_enabled               = (known after apply)
      + key_spec                           = (known after apply)
      + key_usage                          = (known after apply)
      + policy                             = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "kms:*"
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = "arn:aws:iam::894947205914:root"
                        }
                      + Resource  = "*"
                      + Sid       = ""
                    },
                  + {
                      + Action    = [
                          + "kms:ReEncrypt*",
                          + "kms:GenerateDataKey*",
                          + "kms:Encrypt",
                          + "kms:DescribeKey",
                          + "kms:Decrypt",
                        ]
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = [
                              + "arn:aws:iam::894947205914:role/ryandbrown_worker_iam_role",
                              + "arn:aws:iam::894947205914:role/ryandbrown_idp_iam_role",
                            ]
                        }
                      + Resource  = "*"
                      + Sid       = ""
                    },
                  + {
                      + Action    = [
                          + "kms:GenerateDataKey",
                          + "kms:Decrypt",
                        ]
                      + Effect    = "Allow"
                      + Principal = {
                          + Service = [
                              + "sns.amazonaws.com",
                              + "events.amazonaws.com",
                            ]
                        }
                      + Resource  = "*"
                      + Sid       = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + primary_key_arn                    = (known after apply)
      + tags_all                           = (known after apply)
    }

  # module.kms_keymaker_replica_ue1.aws_sns_topic.kms_events will be created
  + resource "aws_sns_topic" "kms_events" {
      + arn                         = (known after apply)
      + content_based_deduplication = false
      + fifo_topic                  = false
      + id                          = (known after apply)
      + name                        = "ryandbrown-decryption-events"
      + name_prefix                 = (known after apply)
      + owner                       = (known after apply)
      + policy                      = (known after apply)
      + signature_version           = (known after apply)
      + tags_all                    = (known after apply)
      + tracing_config              = (known after apply)
    }

  # module.kms_keymaker_replica_ue1.aws_sns_topic_policy.kms_events will be created
  + resource "aws_sns_topic_policy" "kms_events" {
      + arn    = (known after apply)
      + id     = (known after apply)
      + owner  = (known after apply)
      + policy = (known after apply)
    }

  # module.kms_keymaker_replica_ue1.aws_sns_topic_subscription.kms_ct_sqs will be created
  + resource "aws_sns_topic_subscription" "kms_ct_sqs" {
      + arn                             = (known after apply)
      + confirmation_timeout_in_minutes = 1
      + confirmation_was_authenticated  = (known after apply)
      + endpoint                        = "arn:aws:sqs:us-west-2:894947205914:ryandbrown-kms-ct-events"
      + endpoint_auto_confirms          = false
      + filter_policy_scope             = (known after apply)
      + id                              = (known after apply)
      + owner_id                        = (known after apply)
      + pending_confirmation            = (known after apply)
      + protocol                        = "sqs"
      + raw_message_delivery            = false
      + topic_arn                       = (known after apply)
    }

  # module.kms_keymaker_ue1.aws_cloudwatch_event_rule.decrypt will be destroyed
  # (because aws_cloudwatch_event_rule.decrypt is not in configuration)
  - resource "aws_cloudwatch_event_rule" "decrypt" {
      - arn            = "arn:aws:events:us-east-1:894947205914:rule/ryandbrown-decryption-events" -> null
      - description    = "Capture decryption events" -> null
      - event_bus_name = "default" -> null
      - event_pattern  = jsonencode(
            {
              - detail      = {
                  - eventName         = [
                      - "Decrypt",
                    ]
                  - eventSource       = [
                      - "kms.amazonaws.com",
                    ]
                  - requestParameters = {
                      - encryptionContext = {
                          - context = [
                              - "password-digest",
                              - "pii-encryption",
                            ]
                        }
                    }
                  - resources         = {
                      - ARN = [
                          - "arn:aws:kms:us-east-1:894947205914:key/ace13022-019c-45a7-aa80-c3d0bc7affb4",
                        ]
                    }
                }
              - detail-type = [
                  - "AWS API Call via CloudTrail",
                ]
              - source      = [
                  - "aws.kms",
                ]
            }
        ) -> null
      - id             = "ryandbrown-decryption-events" -> null
      - is_enabled     = true -> null
      - name           = "ryandbrown-decryption-events" -> null
      - tags           = {} -> null
      - tags_all       = {} -> null
    }

  # module.kms_keymaker_ue1.aws_cloudwatch_event_target.sns will be destroyed
  # (because aws_cloudwatch_event_target.sns is not in configuration)
  - resource "aws_cloudwatch_event_target" "sns" {
      - arn            = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events" -> null
      - event_bus_name = "default" -> null
      - id             = "ryandbrown-decryption-events-ryandbrown-sns" -> null
      - rule           = "ryandbrown-decryption-events" -> null
      - target_id      = "ryandbrown-sns" -> null
    }

  # module.kms_keymaker_ue1.aws_kms_alias.login-dot-gov-keymaker-alias will be destroyed
  # (because aws_kms_alias.login-dot-gov-keymaker-alias is not in configuration)
  - resource "aws_kms_alias" "login-dot-gov-keymaker-alias" {
      - arn            = "arn:aws:kms:us-east-1:894947205914:alias/ryandbrown-login-dot-gov-keymaker" -> null
      - id             = "alias/ryandbrown-login-dot-gov-keymaker" -> null
      - name           = "alias/ryandbrown-login-dot-gov-keymaker" -> null
      - target_key_arn = "arn:aws:kms:us-east-1:894947205914:key/ace13022-019c-45a7-aa80-c3d0bc7affb4" -> null
      - target_key_id  = "ace13022-019c-45a7-aa80-c3d0bc7affb4" -> null
    }

  # module.kms_keymaker_ue1.aws_kms_key.login-dot-gov-keymaker will be destroyed
  # (because aws_kms_key.login-dot-gov-keymaker is not in configuration)
  - resource "aws_kms_key" "login-dot-gov-keymaker" {
      - arn                                = "arn:aws:kms:us-east-1:894947205914:key/ace13022-019c-45a7-aa80-c3d0bc7affb4" -> null
      - bypass_policy_lockout_safety_check = false -> null
      - customer_master_key_spec           = "SYMMETRIC_DEFAULT" -> null
      - description                        = "ryandbrown-login-dot-gov-keymaker" -> null
      - enable_key_rotation                = true -> null
      - id                                 = "ace13022-019c-45a7-aa80-c3d0bc7affb4" -> null
      - is_enabled                         = true -> null
      - key_id                             = "ace13022-019c-45a7-aa80-c3d0bc7affb4" -> null
      - key_usage                          = "ENCRYPT_DECRYPT" -> null
      - multi_region                       = false -> null
      - policy                             = jsonencode(
            {
              - Statement = [
                  - {
                      - Action    = "kms:*"
                      - Effect    = "Allow"
                      - Principal = {
                          - AWS = "arn:aws:iam::894947205914:root"
                        }
                      - Resource  = "*"
                      - Sid       = ""
                    },
                  - {
                      - Action    = [
                          - "kms:ReEncrypt*",
                          - "kms:GenerateDataKey*",
                          - "kms:Encrypt",
                          - "kms:DescribeKey",
                          - "kms:Decrypt",
                        ]
                      - Effect    = "Allow"
                      - Principal = {
                          - AWS = [
                              - "arn:aws:iam::894947205914:role/ryandbrown_worker_iam_role",
                              - "arn:aws:iam::894947205914:role/ryandbrown_idp_iam_role",
                            ]
                        }
                      - Resource  = "*"
                      - Sid       = ""
                    },
                  - {
                      - Action    = [
                          - "kms:GenerateDataKey",
                          - "kms:Decrypt",
                        ]
                      - Effect    = "Allow"
                      - Principal = {
                          - Service = [
                              - "sns.amazonaws.com",
                              - "events.amazonaws.com",
                            ]
                        }
                      - Resource  = "*"
                      - Sid       = ""
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> null
      - tags                               = {} -> null
      - tags_all                           = {} -> null
    }

  # module.kms_keymaker_ue1.aws_sns_topic.kms_events will be destroyed
  # (because aws_sns_topic.kms_events is not in configuration)
  - resource "aws_sns_topic" "kms_events" {
      - application_success_feedback_sample_rate = 0 -> null
      - arn                                      = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events" -> null
      - content_based_deduplication              = false -> null
      - fifo_topic                               = false -> null
      - firehose_success_feedback_sample_rate    = 0 -> null
      - http_success_feedback_sample_rate        = 0 -> null
      - id                                       = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events" -> null
      - lambda_success_feedback_sample_rate      = 0 -> null
      - name                                     = "ryandbrown-decryption-events" -> null
      - owner                                    = "894947205914" -> null
      - policy                                   = jsonencode(
            {
              - Id        = "kms_ct_sqs"
              - Statement = [
                  - {
                      - Action    = "SNS:Publish"
                      - Effect    = "Allow"
                      - Principal = {
                          - Service = "events.amazonaws.com"
                        }
                      - Resource  = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events"
                      - Sid       = "KMSCTSQS"
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> null
      - signature_version                        = 0 -> null
      - sqs_success_feedback_sample_rate         = 0 -> null
      - tags                                     = {} -> null
      - tags_all                                 = {} -> null
    }

  # module.kms_keymaker_ue1.aws_sns_topic_policy.kms_events will be destroyed
  # (because aws_sns_topic_policy.kms_events is not in configuration)
  - resource "aws_sns_topic_policy" "kms_events" {
      - arn    = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events" -> null
      - id     = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events" -> null
      - owner  = "894947205914" -> null
      - policy = jsonencode(
            {
              - Id        = "kms_ct_sqs"
              - Statement = [
                  - {
                      - Action    = "SNS:Publish"
                      - Effect    = "Allow"
                      - Principal = {
                          - Service = "events.amazonaws.com"
                        }
                      - Resource  = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events"
                      - Sid       = "KMSCTSQS"
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> null
    }

  # module.kms_keymaker_ue1.aws_sns_topic_subscription.kms_ct_sqs will be destroyed
  # (because aws_sns_topic_subscription.kms_ct_sqs is not in configuration)
  - resource "aws_sns_topic_subscription" "kms_ct_sqs" {
      - arn                             = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events:90acbc5a-c0a6-4b4d-a635-9043a1d0ad4f" -> null
      - confirmation_timeout_in_minutes = 1 -> null
      - confirmation_was_authenticated  = true -> null
      - endpoint                        = "arn:aws:sqs:us-west-2:894947205914:ryandbrown-kms-ct-events" -> null
      - endpoint_auto_confirms          = false -> null
      - id                              = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events:90acbc5a-c0a6-4b4d-a635-9043a1d0ad4f" -> null
      - owner_id                        = "894947205914" -> null
      - pending_confirmation            = false -> null
      - protocol                        = "sqs" -> null
      - raw_message_delivery            = false -> null
      - topic_arn                       = "arn:aws:sns:us-east-1:894947205914:ryandbrown-decryption-events" -> null
    }

  # module.kms_keymaker_uw2.aws_cloudwatch_event_rule.decrypt will be updated in-place
  ~ resource "aws_cloudwatch_event_rule" "decrypt" {
      ~ event_pattern  = jsonencode(
            {
              - detail      = {
                  - eventName         = [
                      - "Decrypt",
                    ]
                  - eventSource       = [
                      - "kms.amazonaws.com",
                    ]
                  - requestParameters = {
                      - encryptionContext = {
                          - context = [
                              - "password-digest",
                              - "pii-encryption",
                            ]
                        }
                    }
                  - resources         = {
                      - ARN = [
                          - "arn:aws:kms:us-west-2:894947205914:key/9ca211bb-6cd9-4d9e-a4bd-0d0415a71bd9",
                        ]
                    }
                }
              - detail-type = [
                  - "AWS API Call via CloudTrail",
                ]
              - source      = [
                  - "aws.kms",
                ]
            }
        ) -> (known after apply)
        id             = "ryandbrown-decryption-events"
        name           = "ryandbrown-decryption-events"
        tags           = {}
        # (5 unchanged attributes hidden)
    }

  # module.kms_keymaker_uw2.aws_kms_alias.login-dot-gov-keymaker-multi-region will be created
  + resource "aws_kms_alias" "login-dot-gov-keymaker-multi-region" {
      + arn            = (known after apply)
      + id             = (known after apply)
      + name           = "alias/ryandbrown-login-dot-gov-keymaker-multi-region"
      + name_prefix    = (known after apply)
      + target_key_arn = (known after apply)
      + target_key_id  = "9ca211bb-6cd9-4d9e-a4bd-0d0415a71bd9"
    }

  # module.kms_keymaker_uw2.aws_kms_key.login-dot-gov-keymaker-multi-region will be created
  + resource "aws_kms_key" "login-dot-gov-keymaker-multi-region" {
      + arn                                = (known after apply)
      + bypass_policy_lockout_safety_check = false
      + customer_master_key_spec           = "SYMMETRIC_DEFAULT"
      + description                        = "ryandbrown-login-dot-gov-keymaker-multi-region"
      + enable_key_rotation                = true
      + id                                 = (known after apply)
      + is_enabled                         = true
      + key_id                             = (known after apply)
      + key_usage                          = "ENCRYPT_DECRYPT"
      + multi_region                       = true
      + policy                             = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "kms:*"
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = "arn:aws:iam::894947205914:root"
                        }
                      + Resource  = "*"
                      + Sid       = ""
                    },
                  + {
                      + Action    = [
                          + "kms:ReEncrypt*",
                          + "kms:GenerateDataKey*",
                          + "kms:Encrypt",
                          + "kms:DescribeKey",
                          + "kms:Decrypt",
                        ]
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = [
                              + "arn:aws:iam::894947205914:role/ryandbrown_worker_iam_role",
                              + "arn:aws:iam::894947205914:role/ryandbrown_idp_iam_role",
                            ]
                        }
                      + Resource  = "*"
                      + Sid       = ""
                    },
                  + {
                      + Action    = [
                          + "kms:GenerateDataKey",
                          + "kms:Decrypt",
                        ]
                      + Effect    = "Allow"
                      + Principal = {
                          + Service = [
                              + "sns.amazonaws.com",
                              + "events.amazonaws.com",
                            ]
                        }
                      + Resource  = "*"
                      + Sid       = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + tags_all                           = {
          + "fisma" = "Q-LG"
        }
    }

  # module.env-runner[0].module.outbound_proxy.module.outboundproxy_cloudwatch_filters.aws_cloudwatch_metric_alarm.squid_denied_alarm will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "squid_denied_alarm" {
        id                        = "ryandbrown-squid-denials"
        tags                      = {}
      ~ treat_missing_data        = "breaching" -> "missing"
        # (17 unchanged attributes hidden)
    }

  # module.env-runner[0].module.outbound_proxy.module.outboundproxy_cloudwatch_filters.aws_cloudwatch_metric_alarm.squid_total_requests will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "squid_total_requests" {
        id                        = "ryandbrown-squid-total-requests"
        tags                      = {}
      ~ treat_missing_data        = "breaching" -> "missing"
        # (17 unchanged attributes hidden)
    }

Plan: 9 to add, 5 to change, 7 to destroy.


```

</details>

File: kms_keymaker/cloudwatch.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                "context": [
+                    "password-digest",
+                    "pii-encryption"
+                ]
+            }
+        },
+        "resources": {
+            "ARN": [
+                "${aws_kms_key.login-dot-gov-keymaker.arn}",
+                "${aws_kms_key.login-dot-gov-keymaker-multi-region.arn}"
```
Comment (henrydrichardson) [inline kms_keymaker/cloudwatch.tf:32]:
```suggestion
                "${aws_kms_key.login_dot_gov_keymaker_multi_region.arn}"
```

File: kms_keymaker_replica/kms.tf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+}
+
+resource "aws_kms_replica_key" "login-dot-gov-keymaker-multi-region-replica" {
+  description         = "${var.env_name}-login-dot-gov-keymaker-multi-region"
+  policy              = data.aws_iam_policy_document.kms.json
+  primary_key_arn     = var.primary_key_arn
+}
+
+resource "aws_kms_alias" "login-dot-gov-keymaker-alias" {
```
Comment (henrydrichardson) [inline kms_keymaker_replica/kms.tf:64]:
```suggestion
resource "aws_kms_alias" "login_dot_gov_keymaker_alias" {
```

File: kms_keymaker_replica/kms.tf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      type = "Service"
+      identifiers = [
+        "events.amazonaws.com",
+        "sns.amazonaws.com",
+      ]
+    }
+  }
+}
+
+resource "aws_kms_replica_key" "login-dot-gov-keymaker-multi-region-replica" {
```
Comment (henrydrichardson) [inline kms_keymaker_replica/kms.tf:58]:
```suggestion
resource "aws_kms_replica_key" "login_dot_gov_keymaker_multi_region_replica" {
```

File: kms_keymaker_replica/kms.tf [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  description         = "${var.env_name}-login-dot-gov-keymaker-multi-region"
+  policy              = data.aws_iam_policy_document.kms.json
+  primary_key_arn     = var.primary_key_arn
+}
+
+resource "aws_kms_alias" "login-dot-gov-keymaker-alias" {
+  name          = "alias/${var.env_name}-login-dot-gov-keymaker-multi-region"
+  target_key_id = aws_kms_replica_key.login-dot-gov-keymaker-multi-region-replica.key_id
```
Comment (henrydrichardson) [inline kms_keymaker_replica/kms.tf:66]:
```suggestion
  target_key_id = aws_kms_replica_key.login_dot_gov_keymaker_multi_region_replica.key_id
```

File: kms_keymaker_replica/variables.tf [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  description = "Environment name"
+}
+
+variable "sqs_queue_arn" {
+  description = "ARN of the SQS queue used as the CloudWatch event target."
+}
+
+variable "primary_key_arn" {
+  description = "ARN of the multi-region kms key."
+}
```
Comment (henrydrichardson) [inline kms_keymaker_replica/variables.tf:16]: Just a nitpick on types:

```suggestion
variable "ec2_kms_arns" {
  type        = list(string)
  default     = []
  description = "ARN(s) of EC2 roles permitted access to KMS"
}

variable "env_name" {
  type        = string
  description = "Environment name"
}

variable "sqs_queue_arn" {
  type        = string
  description = "ARN of the SQS queue used as the CloudWatch event target."
}

variable "primary_key_arn" {
  type        = string
  description = "ARN of the multi-region kms key."
}

File: kms_keymaker/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,3 @@
+output "multi_region_primary_key_arn" {
+  value = aws_kms_key.login-dot-gov-keymaker-multi-region.arn
```
Comment (henrydrichardson) [inline kms_keymaker/outputs.tf:2]:
```suggestion
  value = aws_kms_key.login_dot_gov_keymaker_multi_region.arn
```

File: kms_keymaker/kms.tf [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+resource "aws_kms_key" "login-dot-gov-keymaker-multi-region" {
+  multi_region        = true
+  enable_key_rotation = true
+  description         = "${var.env_name}-login-dot-gov-keymaker-multi-region"
+  policy              = data.aws_iam_policy_document.kms.json
+}
+
+resource "aws_kms_alias" "login-dot-gov-keymaker-multi-region" {
+  name          = "alias/${var.env_name}-login-dot-gov-keymaker-multi-region"
+  target_key_id = aws_kms_key.login-dot-gov-keymaker-multi-region.key_id
```
Comment (henrydrichardson) [inline kms_keymaker/kms.tf:78]:
```suggestion
  target_key_id = aws_kms_key.login_dot_gov_keymaker_multi_region.key_id
```

File: kms_keymaker/kms.tf [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+resource "aws_kms_key" "login-dot-gov-keymaker-multi-region" {
+  multi_region        = true
+  enable_key_rotation = true
+  description         = "${var.env_name}-login-dot-gov-keymaker-multi-region"
+  policy              = data.aws_iam_policy_document.kms.json
+}
+
+resource "aws_kms_alias" "login-dot-gov-keymaker-multi-region" {
```
Comment (henrydrichardson) [inline kms_keymaker/kms.tf:76]:
```suggestion
resource "aws_kms_alias" "login_dot_gov_keymaker_multi_region" {
```

File: kms_keymaker/kms.tf [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  description         = "${var.env_name}-login-dot-gov-keymaker"
+  policy              = data.aws_iam_policy_document.kms.json
+}
+
+resource "aws_kms_alias" "login-dot-gov-keymaker-alias" {
+  name          = "alias/${var.env_name}-login-dot-gov-keymaker"
+  target_key_id = aws_kms_key.login-dot-gov-keymaker.key_id
+}
+
+resource "aws_kms_key" "login-dot-gov-keymaker-multi-region" {
```
Comment (henrydrichardson) [inline kms_keymaker/kms.tf:69]: Just a nitpick, but we should migrate the new key to an underscore format to mimic the bulk of our terraform code.
```suggestion
resource "aws_kms_key" "login_dot_gov_keymaker_multi_region" {
```

File: kms_keymaker/README.md [hunk 10]
```diff
@@ -1,6 +1,6 @@
 # `kms_keymaker`
 
-This Terraform module will generate a KMS key + alias for use by the Login.gov IdP application, along with a CloudWatch event monitor, and its target (an SNS topic/subscription). It is best utilized alongside the `kms_log` module, which creates the SQS queue (whose ARN is a required variable in this module) that SNS sends messages to.
+This Terraform module will generate a KMS keys + aliases for use by the Login.gov IdP application, along with a CloudWatch event monitor, and its target (an SNS topic/subscription). It is best utilized alongside the `kms_log` module, which creates the SQS queue (whose ARN is a required variable in this module) that SNS sends messages to.
```
Comment (henrydrichardson) [inline kms_keymaker/README.md:3]: Just a nitpick!

```suggestion
This Terraform module will generate KMS keys + aliases for use by the Login.gov IdP application, along with a CloudWatch event monitor, and its target (an SNS topic/subscription). It is best utilized alongside the `kms_log` module, which creates the SQS queue (whose ARN is a required variable in this module) that SNS sends messages to.
```

**Review by henrydrichardson — COMMENTED** [review]
Overall this looks good. Just a couple of nitpicks.

File: kms_keymaker_multiregion_primary/README.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+This Terraform module will generate multiregion KMS keys + aliases for use by the Login.gov IdP application, along with a CloudWatch event monitor, and its target (an SNS topic/subscription). It is best utilized alongside the `kms_log` module, which creates the SQS queue (whose ARN is a required variable in this module) that SNS sends messages to.
+
+These resources were, previously, all included within `kms_log` in a more monolithic design; they have been split out separately to allow for multi-region KMS keys and associated resources.
+
+## Example
+
+```hcl
+module "kms_keymaker_uw2" {
+  source = "github.com/18F/identity-terraform//kms_keymaker_multiregion?ref=main"
```
Comment (henrydrichardson) [inline kms_keymaker_multiregion_primary/README.md:11]: I'm back with nitpicks!

```suggestion
  source = "github.com/18F/identity-terraform//kms_keymaker_multiregion_primary?ref=main"
```

File: kms_keymaker_multiregion_primary/sns.tf [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    sid = "KMSCTSQS"
+    actions = [
+      "SNS:Publish",
+    ]
+
+    #    condition {
+    #      test     = "StringLike"
+    #      variable = "aws:SourceArn"
+    #      values = [aws_cloudwatch_event_rule.decrypt.arn]
+    #    }
```
Comment (henrydrichardson) [inline kms_keymaker_multiregion_primary/sns.tf:14]: Is this comment necessary or can it be pruned?

File: kms_keymaker_multiregion_primary/variables.tf [hunk 13]
```diff
@@ -0,0 +1,12 @@
+variable "ec2_kms_arns" {
+  default     = []
+  description = "ARN(s) of EC2 roles permitted access to KMS"
+}
+
+variable "env_name" {
+  description = "Environment name"
+}
```
Comment (henrydrichardson) [inline kms_keymaker_multiregion_primary/variables.tf:8]: Typing.

```suggestion
variable "env_name" {
  type        = string
  description = "Environment name"
}
```

File: kms_keymaker_multiregion_primary/variables.tf [hunk 14]
```diff
@@ -0,0 +1,12 @@
+variable "ec2_kms_arns" {
+  default     = []
+  description = "ARN(s) of EC2 roles permitted access to KMS"
+}
```
Comment (henrydrichardson) [inline kms_keymaker_multiregion_primary/variables.tf:4]: Typing:

```suggestion
variable "ec2_kms_arns" {
  default     = []
  type        = list(string)
  description = "ARN(s) of EC2 roles permitted access to KMS"
}
```

File: kms_keymaker_multiregion_primary/variables.tf [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+
+variable "env_name" {
+  description = "Environment name"
+}
+
+variable "sqs_queue_arn" {
+  description = "ARN of the SQS queue used as the CloudWatch event target."
+}
```
Comment (henrydrichardson) [inline kms_keymaker_multiregion_primary/variables.tf:12]: Typing
```suggestion
variable "sqs_queue_arn" {
  type        = string
  description = "ARN of the SQS queue used as the CloudWatch event target."
}
```

File: kms_keymaker_multiregion_replica/variables.tf [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+  description = "Environment name"
+}
+
+variable "sqs_queue_arn" {
+  description = "ARN of the SQS queue used as the CloudWatch event target."
+}
+
+variable "primary_key_arn" {
+  description = "ARN of the multi-region kms key."
+}
```
Comment (henrydrichardson) [inline kms_keymaker_multiregion_replica/variables.tf:16]: Typing:

```suggestion
variable "ec2_kms_arns" {
  default     = []
  type        = list(string)
  description = "ARN(s) of EC2 roles permitted access to KMS"
}

variable "env_name" {
  type        = string
  description = "Environment name"
}

variable "sqs_queue_arn" {
  type        = string
  description = "ARN of the SQS queue used as the CloudWatch event target."
}

variable "primary_key_arn" {
  type        = string
  description = "ARN of the multi-region kms key."
}
```

**Review by henrydrichardson — APPROVED** [review]
LGTM! 

I have a few nitpicks but functionally they have no impact.

**Pull request merged**